### PR TITLE
BFD-1596: code change to work with new schema: snf_claims_new and snf_claim_lines_new

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-1596](https://jira.cms.gov/browse/BFD-1596)

**User Story or Bug Summary:**
BFD code changes to use new, modified schema tables, _snf_claims_new_ and _snf_claim_lines_new_.

### What Does This PR Do?
This PR changes how BFD services provide SNF claims data to requesters. The changes germane to this functionality:
* modify the Excel workbook that drives the RifLayoutsProcessor generation of ORM models; in particular, modifies the SNFClaims worksheet to match new db table content.
* Maintains support for yet-to-be migrated claims tables while at the same time, allowing for changes to the SNF claims table(s) data types and table structure.

### What Should Reviewers Watch For?
Reviewers should:
* verify that code changes to match SNF claims data table structure (i.e., code matches schema)
* check comments to ensure they adequately explain the ongoing migration effort, and that changes associated with this PR are directed towards to producing valid FHIR responses for SNF claims.
* Adequately identifies code changes that may be considered transient (i.e, places where code will be reverted when all BFD data tables have been migrated and cod cleanup becomes necessary).
* Assert that code changes and associated db schema changes do not introduce a regression and continue to produce expected FHIR responses for both stu3 and R4.

If you're reviewing this PR, please check for these things in particular:

* check FHIR responses match current Master API codebase responses, especially those that include SNF claims data.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:
    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

* Database migrations tied to BFD-1595 (https://github.com/CMSgov/beneficiary-fhir-data/tree/cmac/BFD-1595-SNF-Claims-Migration)

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* https://github.com/CMSgov/beneficiary-fhir-data/pull/1027

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java [hunk 1]
```diff
@@ -311,7 +332,9 @@ public String getLineEntityLineNumberField() {
    *     that should be used for the identifying yearMonth, if any
    */
   public String getEntityBeneficiaryMonthlyField() {
-    if (!isBeneficiaryEntity) throw new IllegalStateException();
+    if (!isBeneficiaryEntity) {
+      throw new IllegalStateException();
+    }
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java:337]: these types of changes seem to be the only type in this file. any reason why this is changing on this PR?

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 2]
```diff
@@ -141,6 +141,23 @@ public static Long parseLong(String longText) {
...
 
+  /**
+   * @param shortText the number string to parse
+   * @return the specified text parsed into an {@link Long}
+   */
+  public static short parseShort(String shortText) {
+    /*
+     * Might seem silly to pull this out, but it makes the code a bit easier
+     * to read, and ensures that this parsing is standardized.
+     */
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:152]: I suggest getting rid of this low-value comment here (I know that it got copy pasted from the integer one)

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 3]
```diff
@@ -141,6 +141,23 @@ public static Long parseLong(String longText) {
     }
   }
 
+  /**
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:144]: Can you put a description here as that will be in the checkstyle rules later

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 4]
```diff
@@ -164,11 +177,16 @@ public static Long parseLong(String longText) {
...
   public static Optional<Long> parseOptionalLong(String longText) {
-    if (longText.isEmpty()) {
-      return Optional.empty();
-    } else {
-      return Optional.of(parseLong(longText));
-    }
+    return longText.isEmpty() ? Optional.empty() : Optional.of(parseLong(longText));
+  }
+
+  /**
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:183]: description

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 5]
```diff
@@ -295,11 +303,7 @@ public static Instant parseTimestamp(String timestampText) {
    *     Optional if not
    */
   public static Optional<Instant> parseOptionalTimestamp(String timestampText) {
-    if (timestampText.isEmpty()) {
-      return Optional.empty();
-    } else {
-      return Optional.of(parseTimestamp(timestampText));
-    }
+    return timestampText.isEmpty() ? Optional.empty() : Optional.of(parseTimestamp(timestampText));
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:306]: I see a lot of these changes in here -- I think we ought to add this to a future style guide discussion before changing them everywhere. It's a personal preference that we should discuss once and adopt a standard for. Without that I think changing them everywhere is premature.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 6]
```diff
@@ -165,7 +165,7 @@ public static IdDt buildPatientId(Beneficiary beneficiary) {
    *     Beneficiary}
    */
   public static IdDt buildPatientId(Long beneficiaryId) {
-    return new IdDt(Patient.class.getSimpleName(), beneficiaryId.toString());
+    return new IdDt(Patient.class.getSimpleName(), beneficiaryId);
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:168]: I remember this change from a previous PR. Why the change back?

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 7]
```diff
@@ -404,6 +404,26 @@ static Identifier createClaimIdentifier(
     return identifier;
   }
 
+  /**
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:407]: description

**Review by christrotteradhoc — COMMENTED** [review]
nothing major, but a few questions

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java (hunk 1 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java:337]: Apart from being grievously affronted by the lack of brackets? Personally. I believe this is an absolute line in the sand as far as code is concerned. Coming from background in other langs (espec c/c++) I have seen too many errors because devs miss things when not absolutely scoped. So my linter barks at this sort of stuff...

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 2 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:152]: ok, if I remove this, I think we should remove them all; there's not a whole lot of value to the comment, and there's nothing wrong with code that handles particular inputs.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 5 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:306]: I 'sort of' agree.....it goes back to my firm belief of brackets & scoping (and another one of multiple return statements). So I could change the code to include the brackets which will result in 5 ilnes of change:

if (timestampText.isEmpty()) {
    return Optional.empty();
} else {
    return Optional.of(parseTimestamp(timestampText));
}

Now as a lazy developer, I'd rather have one line than 5!

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 6 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:168]: This is the proverbial "more than one way to skin a cat". Now for something like this I prefer to let the class itself handle things; to wit, here are the 2 pertinent constructors:

IdDt(String theResourceType, Long theIdPart)
IdDt(String theResourceType, String theId)

So in this case, I think it's better to pass the Long and let the class handle it rather than do some gyration in the caller to 'fit it in'.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java (hunk 1 above)
Comment (christrotteradhoc) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java:337]: I guess that last part is what I don't understand -- this class has no changes other than this cleanup that I can see.

I'm all for cleaning up stuff but I think the approach should be:

1. propose a style rule
2. get buy-in from the team and document the result in the style guide
3. modify the formatter and linter accordingly so that the change can be applied globally and not regress later
4. apply the change to the entire codebase at once in a single standalone PR

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 6 above)
Comment (christrotteradhoc) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:167]: This is probably academic at this point since I agree with this change. However this is what it looked like originally and you modified it at the very end of 1532 because you found something in your testing. I just want to be certain that there is not something here of concern as I didn't understand originally why you wanted to change it.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 5 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:284]: agreed; however I did have to introduce code (new parser for bigint, smallint, and integer); and I will always use the ternary operator for these simple if-then-else. So my conundrum was add my changes using ternary and then field questions as to why didn't I use a verbose if-then-else.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 6 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:167]: no, did not find anything other than doing some QC of code and I knew from previous work (BFD1076) that the FHIR IdDt class supported Long.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java [hunk 8]
```diff
@@ -1892,6 +1910,20 @@ private static List<RifField> createDetailsForAdditionalDatabaseFields(
         addlDatabaseFields.add(lastUpdated);
         continue;
       }
+      if (additionalDatabaseField.contentEquals("BENE_ID_NUMERIC")) {
+        RifField lastUpdated =
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:1914]: I see the same error in the other fields, but this var shouldnt be named lastUpdated since this is bene_id_numberic.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java [hunk 9]
```diff
@@ -2203,26 +2235,27 @@ private static TypeName selectJavaFieldType(
...
-    else if (type == RifColumnType.TIMESTAMP && columnLength.orElse(0) == 20)
-      return ClassName.get(Instant.class);
+    if (type == RifColumnType.CHAR) {
+      if (columnLength.orElse(Integer.MAX_VALUE) == 1) {
+        return isColumnOptional ? ClassName.get(Character.class) : TypeName.CHAR;
+      } else {
+        return ClassName.get(String.class);
+      }
+    } else if (type == RifColumnType.DATE) return ClassName.get(LocalDate.class);
+    else if (type == RifColumnType.TIMESTAMP) return ClassName.get(Instant.class);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:2245]: Here's some cursed non-bracketed if statements you could have justification to fix 😄

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java [hunk 10]
```diff
@@ -2203,26 +2235,27 @@ private static TypeName selectJavaFieldType(
...
       return ClassName.get(BigDecimal.class);
     else if (type == RifColumnType.NUM
         && columnScale.orElse(Integer.MAX_VALUE) == 0
         && !isColumnOptional) return TypeName.INT;
-    else if (type == RifColumnType.NUM
-        && columnScale.orElse(Integer.MAX_VALUE) == 0
-        && isColumnOptional) return ClassName.get(Integer.class);
-    else throw new IllegalArgumentException("Unhandled field type: " + type.name());
+    else if (type == RifColumnType.SMALLINT) {
+      return isColumnOptional ? ClassName.get(Short.class) : TypeName.SHORT;
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:2252]: What is this even doing? This method could particularly use an @returns in the doc cause I don't know whats even happening here. So we return "Short" or "short" based on.. the column being optional? Not a condemnation of this change since its following the convention of the method, but its just written confusingly in the first place.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 11]
```diff
@@ -141,6 +137,19 @@ public static Long parseLong(String longText) {
     }
   }
 
+  /**
+   * @param shortText the number string to parse
+   * @return the specified text parsed into an {@link Long}
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:142]: Should be updated to say short instead of long

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java [hunk 12]
```diff
@@ -178,12 +188,8 @@ public static Optional<Long> parseOptionalLong(String longText) {
    * @return the specified text parsed into a {@link BigDecimal}
    */
   public static BigDecimal parseDecimal(String decimalText) {
-    /*
-     * Might seem silly to pull this out, but it makes the code a bit easier
-     * to read, and ensures that this parsing is standardized.
-     */
     if (decimalText.isEmpty()) {
-      return new BigDecimal(0);
+      return BigDecimal.valueOf(0);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:192]: 🍰  Can also use BigDecimal.ZERO

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 5 above)
Comment (lsmitchell) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:284]: IMO ternaries are fine for simple logic splits like this, but is certainly a good thing to put on the list of things to discuss as part of a greater style guide.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java [hunk 13]
```diff
@@ -398,8 +398,18 @@ private <T> List<T> findClaimTypeByPatient(
...
+    // we may need to change the expected type for beneficiaryId in the entity Predicate.
+    // Once all claims have been migrated we can modify the ClaimTypeV2 to specifically
+    // return a long data type and this extra data type checking can be removed.
+    java.lang.Class javaClass = claimType.getEntityBeneficiaryIdAttribute().getJavaType();
+
     Predicate wherePredicate =
-        builder.equal(root.get(claimType.getEntityBeneficiaryIdAttribute()), patientId);
+        builder.equal(
+            root.get(claimType.getEntityBeneficiaryIdAttribute()),
+            javaClass.getName().equals("long") ? Long.parseLong(patientId) : patientId);
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java:411]: I think getJavaType was returning class.getName under certain conditions, wasnt it? Will that return "Long" or the lowercase version? (And if it returns the same value, why do we return the same value in two different ways within getJavaType?)

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/SNFClaimTransformer.java [hunk 14]
```diff
@@ -345,7 +345,10 @@ private static ExplanationOfBenefit transformClaim(SNFClaim claimGroup) {
           claimLine.getTotalChargeAmount(),
           claimLine.getNonCoveredChargeAmount(),
           BigDecimal.valueOf(claimLine.getUnitCount()),
-          claimLine.getNationalDrugCodeQuantity(),
+          claimLine.getNationalDrugCodeQuantity().isPresent()
+              ? Optional.of(
+                  BigDecimal.valueOf(claimLine.getNationalDrugCodeQuantity().get().longValue()))
+              : Optional.empty(),
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/SNFClaimTransformer.java:351]: IMO, This is near the line of complexity where i'd recommend breaking it out of a ternary and into a statement, then pass the result of the statement to this line.

**Review by lsmitchell — COMMENTED** [review]
Given the larger code changes in RifLayoutsProcessor, I'd be interested in ensuring the new code changes are properly covered by tests; could you run coverage on that class at least and ensure there are tests for the various code paths in the method(s) you touched, particularly on the new code?

I like a lot of the cleanup in this PR! Hopefully future style guide discussion and work can calcify and automate some similar styles so we can clean this up more thoroughly in the codebase.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java (hunk 13 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java:411]: Depends on the actual db column type AND whether the db column can be nullable. There is a very simple rule to be applied here:

If the db column is defined as an integral data data **AND that column cannot be null**, then use a native primitive; in this case 99% of the time, beneficiaryId is not nullable and therefore ORM models need to treat it as a long, period.

However, we have one table, medicare_benficiaryid_history where bene_id is defined as nullable; so for that we need a Long data type.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 12 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:192]: good call (and more efficient). changing to use .ZERO.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java (hunk 13 above)
Comment (lsmitchell) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java:411]: Thanks for context on the temp migration code! Looking forward to this being simplified once we're done with the migrations. (Maybe the temporary code should be denoted as such for easier removal later?)

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/SNFClaimTransformer.java (hunk 14 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/SNFClaimTransformer.java:351]: agreed; will break apart retrieval of the long value prior to the ternary.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java (hunk 10 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:2252]: So the TypeName that it returns is a Java Poet ORM data type to be applied to that column in the entity model. I've added a return to try to explain a bit more. And yes your are right on many counts....it is confusing (to say the least) and the isColumnOptional drives whether the column can be a primitive or a Java class.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java (hunk 9 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:2245]: yea, already got beat up for fixing some of the others; I basically operated on the premise that if I had to modify a method, I was not going to perpetuate bad coding practices. I'll take a further look at this one, to see it meets my gag criteria.

Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java:2245]: I did add those lines to my gag reflex fixing on non-bracketed code.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java (hunk 4 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/parse/RifParsingUtils.java:175]: added description.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java (hunk 9 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java]: As an aside, a comment on your request to add tests coverage.....this is where things could get really tricky and really create some scope creep. Where you asked to add test coverage, we run into the proverbial chicken-egg; the RifLayoutsProcessor is part of the bfd-model-codegen which not only doesn't have any tests, it lives only within the constraints of maven to create ORM models based on an Excel spreadsheet. I'd be in favor of a separate ticket to implement some sort of testing, but literally it'll be bootstrapping an entire test system not just adding a test or two to assert some newly added or changed code.

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java (hunk 1 above)
Comment (christrotteradhoc) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java:337]: what's the plan on this comment? Looks like all other comments resolved and this is about ready to go

Comment (mackec1) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/MappingSpec.java]: reverted back to no brackets.

**Review by christrotteradhoc — DISMISSED** [review]

File: apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java (hunk 9 above)
Comment (lsmitchell) [reply apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java]: Ok, we should capture this as tech debt then.

**Review by lsmitchell — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]
